java/core/src/main/java/com/redhat/rhn/domain/errata/ErrataFactory.java

```diff
@@ -145,44 +144,52 @@ else if (identifier.length() > 4) {
                 }
             }
             if (retval.isEmpty()) {
-                prefix = identifier.substring(0, 4);
-                if (prefix.matches("RH.A")) {
-                    StringTokenizer strtok = new StringTokenizer(identifier, "-");
-                    StringBuilder buf = new StringBuilder();
-                    boolean foundFirst = false;
-                    while (strtok.hasMoreTokens()) {
-                        buf.append(strtok.nextToken());
-                        if (!foundFirst) {
-                            buf.append("-");
-                            foundFirst = true;
-                        }
-                        else {
-                            if (strtok.hasMoreTokens()) {
-                                buf.append(":");
-                            }
-                        }
-                    }
-                    identifier = buf.toString();
-                    erratas = ErrataFactory.lookupByAdvisoryId(identifier, org);
+                tryLookupByPrefixRHA(identifier, org, retval);
```

Comment:
I'd suggest to make the process more explicit, because the handling of `retval` is kind of confusing now. I would replace `tryLookupByPrefixRHA()`  with `List<Errata> lookupByPrefixRHA()`:

```java
                retval.addAll(lookupByPrefixRHA(identifier, org);
```